### ISSUE TITLE
Option for triggering sensor on tampering

### ISSUE THREAD

**Issue by Locutus73** [mike182uk/homebridge-zwave#18]
Hi @mike182uk, I see you closed my pull request about triggering motion detected event on tampering event because in homebridge-zwave 2.1.0 you implemented a tampering characteristic which doesn't interfere with the main motion detection sensor.
While I think having a separate tampering characteristic is cool, I think we should have at least an option for triggering the main sensor (motion detection, window opened, whatever) when tampering is detected. One of the main applications of these sensors are for security/anti-theft systems (I use the Fibaro for this reason) and the "motion detected" event sets off my Alarm scene. Ok, using Eve I can set off the Alarm scene both on "motion detected" event and on a specific tampering characteristic value, but IMHO triggering the "motion detected" event on tampering is more straight off, easy to understand, easy to see, test and manage with Apple Home app. In the end, in the "not smart world" wired motion sensors with normally closed contacts are often wired with "motion detection" and "tampering" in serial connection (so the alarm is connected with just one wire and sets off if motion or tampering is detected).

So this issue is for requesting a configuration option for optionally triggering the sensor main event (motion, door/window opened, whatever) on tampering.

Please tell me if I can help (with code or whatever you may need).

Thank you in advance.

**Comment by Locutus73** [comment]
Hi @mike182uk, I have to correct myself: I installed homebridge-zwave 2.1 and I **can't** use the tampering characteristic as a condition for an automation with Eve, I can only use the "Motion Detected" event. So it seems that triggering the "Motion Detected" characteristic/event when "Tampering" is detected is mandatory for an effective security/anti-theft system (I patched the code in my local installation for that).
I noticed that I can use many characteristics from weird accessories in automations with Eve, so, maybe it's something in the specific characteristic.

Thank you in advance.
Regards.

**Comment by mike182uk** [comment]
Hey @Locutus73 

The problem with this is it is specific to your use case - It might be that somebody **does not** want to trigger `Motion Detected` when a device is `Tampered`. This seems more of a workaround for a limitation of HomeKit (not being able to create automations based on accessory characteristics changing).

I'm not sure what the best thing to do here is, I would need to do a bit of research to see what we could do (if anything) with this

**Comment by Locutus73** [comment]
I totally agree with you that this is my use case (which I guess won't be totally uncommon) and there may be people not wanting that... my request is for an option in config.json for optionally enabling this feature. Something like "TriggerEventOnTampering": true/false which could be global (like "noChache") or specific for an accessory.
In the end I think motion sensors have two common uses:
- Security/anti-theft system: in this case I guess most people want the automation (so the event) be triggered both by PIR/ContactSensor/Whatever and Tampering, because if someone is tampering the sensor, you want the alarm to be triggered.
- Presence detection i.e. for turning on lights; in this case I guess people is not interested in tampering at all (it could be triggered or not indifferently).

Anyway I agree the decision should be left to the end user (with an option 😉).

IMHO the tricky part is handling correctly the events on/off sequence; let me explain with an example...
- time 0: Tampering goes ON
- time 1: Motion Detection goes ON
- time 2: Tampering goes OFF
- time 3: Motion Detection goes OFF

If we enable something like "TriggerEventOnTampering": true, IMHO the main sensor should report the "excitement status" between time 0 and time 3, so while tampering is on or motion is on; with a very basic implementation it could stay on between time 0 and time 2 (the first OFF disables the "excitement").

As always thank you in advance; I never say thank you enough for your precious work.
Ciao.
